---
title: "DE analysis with Mito-Q/G data"
author: "Mark Ziemann"
date: "`r Sys.Date()`"
output:
  html_document:
    toc: true
theme: cosmo
---

Source: https://github.com/markziemann/mitoG.git
View reports: http://118.138.234.73/public/mel/mitoG/

## Introduction

Here are the list of figures I'm thinking to include in the paper:

a. Venn/UpSet (Fig.40/42 in the report)
b. Rank-rank (Fig.43)
c. Rank-rank or enrichment plots to show specific pathways eg electron transport, gluconeogenesis, NFkB, diabetic nephropathy associated genes
d. MA plot(Fig.23) between Akita+V and Akita+MitoG (or something prettier)
e. heatmap between Akita+V and Akita+MitgoG

And anything else you think is appropriate and that we should include in the paper.

## Methods

Approximately 200 ng of total RNA underwent RNA depletion using the NEBNext rRNA Depletion Kit followed by library construction using an NEBNext Ultra II Directional RNA Library Prep Kit for Illumina (both from New England BioLabs Inc.).
Barcoded libraries underwent Illumina 100 cycle single-read sequencing at the Australian Genome Research Facility, Melbourne, using HiSeq version 4 reagents. 
Reads underwent trimming in Skewer version 0.2.2 [1] in order to remove bases with quality less than Phred 20.
Reads were then mapped to Gencode mouse v25 transcripts with kallisto version 0.46.1 [2].
Tabular data was read into R version 4.0.2 for downstream analysis. 
Transcript level counts were then aggegated to gene level counts.
For each contrast, genes with an average of fewer than 10 counts were excluded.
The resulting count matrix underwent differential analysis with DESeq2 version 1.28.1 [3].
Mouse gene names were mapped to human using data from Ensembl Biomart v101 [4]
Enrichment analysis was performed using mitch v1.0.8 [5] with gene sets from Reactome [6] (obtained 8/Oct/2020).
Additionally, gene sets related to diabetes were obtained from MsigDB v7.2 [7].
Genes and gene sets with false discovery rate adjusted p-values (FDR) < 0.05 were considered statistically significant.

## References

1. Jiang H, Lei R, Ding SW, Zhu S. Skewer: a fast and accurate adapter trimmer for next-generation sequencing paired-end reads. BMC Bioinformatics. 2014 Jun 12;15:182. doi: 10.1186/1471-2105-15-182. PMID: 24925680; PMCID: PMC4074385.

2. Bray NL, Pimentel H, Melsted P, Pachter L. Near-optimal probabilistic RNA-seq quantification. Nat Biotechnol. 2016 May;34(5):525-7. doi: 10.1038/nbt.3519. Epub 2016 Apr 4. Erratum in: Nat Biotechnol. 2016 Aug 9;34(8):888. PMID: 27043002.

3. Love MI, Huber W, Anders S. Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2. Genome Biol. 2014;15(12):550. doi: 10.1186/s13059-014-0550-8. PMID: 25516281; PMCID: PMC4302049.

4. Kinsella RJ, Kähäri A, Haider S, Zamora J, Proctor G, Spudich G, Almeida-King J, Staines D, Derwent P, Kerhornou A, Kersey P, Flicek P. Ensembl BioMarts: a hub for data retrieval across taxonomic space. Database (Oxford). 2011 Jul 23;2011:bar030. doi: 10.1093/database/bar030. PMID: 21785142; PMCID: PMC3170168.

5. Kaspi A, Ziemann M. mitch: multi-contrast pathway enrichment for multi-omics and single-cell profiling data. BMC Genomics. 2020 Jun 29;21(1):447. doi: 10.1186/s12864-020-06856-9. PMID: 32600408; PMCID: PMC7325150.

6. Jassal B, Matthews L, Viteri G, Gong C, Lorente P, Fabregat A, Sidiropoulos K, Cook J, Gillespie M, Haw R, Loney F, May B, Milacic M, Rothfels K, Sevilla C, Shamovsky V, Shorser S, Varusai T, Weiser J, Wu G, Stein L, Hermjakob H, D'Eustachio P. The reactome pathway knowledgebase. Nucleic Acids Res. 2020 Jan 8;48(D1):D498-D503. doi: 10.1093/nar/gkz1031. PMID: 31691815; PMCID: PMC7145712.

7. Liberzon A, Birger C, Thorvaldsdóttir H, Ghandi M, Mesirov JP, Tamayo P. The Molecular Signatures Database (MSigDB) hallmark gene set collection. Cell Syst. 2015 Dec 23;1(6):417-425. doi: 10.1016/j.cels.2015.12.004. PMID: 26771021; PMCID: PMC4707969.

## Libraries

```{r,libs}
suppressPackageStartupMessages({
  library("tidyverse")
  library("reshape2")
  library("DESeq2")
  library("gplots")
  library("eulerr")
  library("mitch")
})
```


## Read in the data

First read in the data.
We can see how many reads were assigned.

```{r,read,fig.height=7,fig.width=7}

tmp <- read.table("3col.tsv",header=F)
x <- acast(tmp, V2~V1, value.var="V3", fun.aggregate = sum)
x <- as.data.frame(x)
dim(x)
head(x,2)
par(mar=c(5,15,5,5))
colSums(x)
barplot(colSums(x),horiz=TRUE,las=2,main="assigned reads")

```

## Aggregate reads to genes

```{r,agg}

x$gene_id <- sapply(  strsplit(rownames(x),"\\|") , "[[" ,2 ) 
x$gene_name <- sapply(  strsplit(rownames(x),"\\|") , "[[" ,6 ) 
x$gene <- paste(x$gene_id, x$gene_name)
x$gene_id = x$gene_name = NULL

xx <- aggregate(. ~ gene, x, sum) 
rownames(xx) <- xx$gene
xx$gene = NULL
xx <- round(xx)

```

## MDS plot  

MDS plot shows similarity between samples across the whole experiment.

```{r,mds,fig.height=7,fig.width=7}

par(mar=c(5,5,5,5))
mydist <- cmdscale(dist(t(xx)))
plot(mydist, xlab="Coordinate 1", ylab="Coordinate 2", type = "n")
text(mydist, labels=colnames(xx), )

```

## Sample sheet

```{r,ss}

ss <- as.data.frame ( colnames(xx) )
ss$geno <- factor(as.integer(grepl("het",ss[,1])))
ss$trt <- sapply(strsplit(ss[,1],"_"),"[[",1)
ss$trt <- sapply(strsplit(ss$trt,"-"),"[[",2)

```

## Contrast 1 test for H2O vs EtOH in wt mice

```{r,de1,fig.height=7,fig.width=7}
NAME="DE1: H2O vs EtOH in wt mice"
ss1 <- subset(ss,geno==0)
ss1 <- subset(ss1,trt=="h2o" | trt=="etoh")
ss1$trt <- factor(ss1$trt,levels=c("h2o","etoh"))

x1 <- as.data.frame(xx[,ss1[,1]  ])
x1 <- x1[which(rowSums(x1)>10),] 

dds <- DESeqDataSetFromMatrix(countData = x1 , colData = ss1, design = ~ trt )
res <- DESeq(dds)
z<- results(res)
vsd <- vst(dds, blind=FALSE)
zz<-cbind(as.data.frame(z),assay(vsd))
dge<-as.data.frame(zz[order(zz$pvalue),])
head(dge,10)
de1 <- dge

sig <- subset(dge,padj<0.05)
sig_up <- rownames(subset(sig,log2FoldChange>0))
sig_dn <- rownames(subset(sig,log2FoldChange<0))
cnt_deg <- nrow(sig)
cnt_up <-length(sig_up)
cnt_dn <- length(sig_dn)


#MDS PLOT
plot(cmdscale(dist(t(x1))), xlab="Coordinate 1", ylab="Coordinate 2", type = "n")
text(cmdscale(dist(t(x1))), labels=colnames(x1), )

# ma plot
HEADING=paste(NAME, cnt_deg, "DGEs", cnt_up, "up", cnt_dn, "dn")
plot(log2(dge$baseMean),dge$log2FoldChange,main=HEADING,col="gray",pch=19,cex=0.5,xlab="log2(base mean)",ylab="log2(fold change)")
points(log2(sig$baseMean),sig$log2FoldChange,col="red",pch=19,cex=0.5)

# heatmap
if (cnt_deg>50) {cnt_deg=50}
if (cnt_deg>2) {
  mycols <- gsub("h2o","yellow",ss1$trt)
  mycols <- gsub("etoh","orange",mycols)
  colfunc <- colorRampPalette(c("blue", "white", "red"))
  x1 <- x1/colSums(x1)*1e6
  mx <- as.matrix(x1[which(rownames(x1) %in% rownames(dge)[seq_len(cnt_deg)]),])
  heatmap.2(mx,trace="none",scale="row",margin=c(10,15), 
    ColSideColors =mycols , col=colfunc(25),cexRow = 0.7 )
}

```


## Contrast 2 test for H2O vs mitoQ in wt mice

```{r,de2,fig.height=7,fig.width=7}
NAME="DE2: H2O vs mitoQ in wt mice"
ss1 <- subset(ss,geno==0)
ss1 <- subset(ss1,trt=="h2o" | trt=="mitoQ")
ss1$trt <- factor(ss1$trt,levels=c("h2o","mitoQ"))

x1 <- as.data.frame(xx[,ss1[,1]  ])
x1 <- x1[which(rowSums(x1)>10),]

dds <- DESeqDataSetFromMatrix(countData = x1 , colData = ss1, design = ~ trt )
res <- DESeq(dds)
z<- results(res)
vsd <- vst(dds, blind=FALSE)
zz<-cbind(as.data.frame(z),assay(vsd))
dge<-as.data.frame(zz[order(zz$pvalue),])
head(dge,10)
de2 <- dge

sig <- subset(dge,padj<0.05)
sig_up <- rownames(subset(sig,log2FoldChange>0))
sig_dn <- rownames(subset(sig,log2FoldChange<0))
cnt_deg <- nrow(sig)
cnt_up <-length(sig_up)
cnt_dn <- length(sig_dn)


#MDS PLOT
plot(cmdscale(dist(t(x1))), xlab="Coordinate 1", ylab="Coordinate 2", type = "n")
text(cmdscale(dist(t(x1))), labels=colnames(x1), )

# ma plot
HEADING=paste(NAME, cnt_deg, "DGEs", cnt_up, "up", cnt_dn, "dn")
plot(log2(dge$baseMean),dge$log2FoldChange,main=HEADING,col="gray",pch=19,cex=0.5,xlab="log2(base mean)",ylab="log2(fold change)")
points(log2(sig$baseMean),sig$log2FoldChange,col="red",pch=19,cex=0.5)

# heatmap
if (cnt_deg>50) {cnt_deg=50}
if (cnt_deg>2) {
  mycols <- gsub("h2o","yellow",ss1$trt)
  mycols <- gsub("mitoQ","orange",mycols)
  colfunc <- colorRampPalette(c("blue", "white", "red"))
  x1 <- x1/colSums(x1)*1e6
  mx <- as.matrix(x1[which(rownames(x1) %in% rownames(dge)[seq_len(cnt_deg)]),])
  heatmap.2(mx,trace="none",scale="row",margin=c(10,15),
    ColSideColors =mycols , col=colfunc(25),cexRow = 0.7 )
}

```

## Contrast 3 test for EtOH vs MitoG in wt mice


```{r,de3,fig.height=7,fig.width=7}
NAME="DE3: EtOH vs mitoG in wt mice"
ss1 <- subset(ss,geno==0)
ss1 <- subset(ss1,trt=="etoh" | trt=="mitoG")
ss1$trt <- factor(ss1$trt,levels=c("etoh","mitoG"))

x1 <- as.data.frame(xx[,ss1[,1]  ])
x1 <- x1[which(rowSums(x1)>10),]

dds <- DESeqDataSetFromMatrix(countData = x1 , colData = ss1, design = ~ trt )
res <- DESeq(dds)
z<- results(res)
vsd <- vst(dds, blind=FALSE)
zz<-cbind(as.data.frame(z),assay(vsd))
dge<-as.data.frame(zz[order(zz$pvalue),])
head(dge,10)
de3 <- dge

sig <- subset(dge,padj<0.05)
sig_up <- rownames(subset(sig,log2FoldChange>0))
sig_dn <- rownames(subset(sig,log2FoldChange<0))
cnt_deg <- nrow(sig)
cnt_up <-length(sig_up)
cnt_dn <- length(sig_dn)


#MDS PLOT
plot(cmdscale(dist(t(x1))), xlab="Coordinate 1", ylab="Coordinate 2", type = "n")
text(cmdscale(dist(t(x1))), labels=colnames(x1), )

# ma plot
HEADING=paste(NAME, cnt_deg, "DGEs", cnt_up, "up", cnt_dn, "dn")
plot(log2(dge$baseMean),dge$log2FoldChange,main=HEADING,col="gray",pch=19,cex=0.5,xlab="log2(base mean)",ylab="log2(fold change)")
points(log2(sig$baseMean),sig$log2FoldChange,col="red",pch=19,cex=0.5)

# heatmap
if (cnt_deg>50) {cnt_deg=50}
if (cnt_deg>2) {
  mycols <- gsub("etoh","yellow",ss1$trt)
  mycols <- gsub("mitoG","orange",mycols)
  colfunc <- colorRampPalette(c("blue", "white", "red"))
  x1 <- x1/colSums(x1)*1e6
  mx <- as.matrix(x1[which(rownames(x1) %in% rownames(dge)[seq_len(cnt_deg)]),])
  heatmap.2(mx,trace="none",scale="row",margin=c(10,15),
    ColSideColors =mycols , col=colfunc(25),cexRow = 0.7 )
}

```

## Contrast 4 test for H2O vs EtOH in het mice

```{r,de4,fig.height=7,fig.width=7}
NAME="DE4: H2O vs EtOH het mice"
ss1 <- subset(ss,geno==1)
ss1 <- subset(ss1,trt=="h20" | trt=="etoh")
ss1$trt <- factor(ss1$trt,levels=c("h20","etoh"))

x1 <- as.data.frame(xx[,ss1[,1]  ])
x1 <- x1[which(rowSums(x1)>10),]

dds <- DESeqDataSetFromMatrix(countData = x1 , colData = ss1, design = ~ trt )
res <- DESeq(dds)
z<- results(res)
vsd <- vst(dds, blind=FALSE)
zz<-cbind(as.data.frame(z),assay(vsd))
dge<-as.data.frame(zz[order(zz$pvalue),])
head(dge,10)
de4 <- dge

sig <- subset(dge,padj<0.05)
sig_up <- rownames(subset(sig,log2FoldChange>0))
sig_dn <- rownames(subset(sig,log2FoldChange<0))
cnt_deg <- nrow(sig)
cnt_up <-length(sig_up)
cnt_dn <- length(sig_dn)


#MDS PLOT
plot(cmdscale(dist(t(x1))), xlab="Coordinate 1", ylab="Coordinate 2", type = "n")
text(cmdscale(dist(t(x1))), labels=colnames(x1), )

# ma plot
HEADING=paste(NAME, cnt_deg, "DGEs", cnt_up, "up", cnt_dn, "dn")
plot(log2(dge$baseMean),dge$log2FoldChange,main=HEADING,col="gray",pch=19,cex=0.5,xlab="log2(base mean)",ylab="log2(fold change)")
points(log2(sig$baseMean),sig$log2FoldChange,col="red",pch=19,cex=0.5)

# heatmap
if (cnt_deg>50) {cnt_deg=50}
if (cnt_deg>2) {
  mycols <- gsub("h20","yellow",ss1$trt)
  mycols <- gsub("etoh","orange",mycols)
  colfunc <- colorRampPalette(c("blue", "white", "red"))
  x1 <- x1/colSums(x1)*1e6
  mx <- as.matrix(x1[which(rownames(x1) %in% rownames(dge)[seq_len(cnt_deg)]),])
  heatmap.2(mx,trace="none",scale="row",margin=c(10,15),
    ColSideColors =mycols , col=colfunc(25),cexRow = 0.7 )
}

```


## Contrast 5 test for H2O vs mitoQ in Het mice

```{r,de5,fig.height=7,fig.width=7}
NAME="DE5: H2O vs mitoQ in het mice"
ss1 <- subset(ss,geno==1)
ss1 <- subset(ss1,trt=="h20" | trt=="mitoQ")
ss1$trt <- factor(ss1$trt,levels=c("h20","mitoQ"))

x1 <- as.data.frame(xx[,ss1[,1]  ])
x1 <- x1[which(rowSums(x1)>10),]

dds <- DESeqDataSetFromMatrix(countData = x1 , colData = ss1, design = ~ trt )
res <- DESeq(dds)
z<- results(res)
vsd <- vst(dds, blind=FALSE)
zz<-cbind(as.data.frame(z),assay(vsd))
dge<-as.data.frame(zz[order(zz$pvalue),])
head(dge,10)
de5 <- dge

sig <- subset(dge,padj<0.05)
sig_up <- rownames(subset(sig,log2FoldChange>0))
sig_dn <- rownames(subset(sig,log2FoldChange<0))
cnt_deg <- nrow(sig)
cnt_up <-length(sig_up)
cnt_dn <- length(sig_dn)


#MDS PLOT
plot(cmdscale(dist(t(x1))), xlab="Coordinate 1", ylab="Coordinate 2", type = "n")
text(cmdscale(dist(t(x1))), labels=colnames(x1), )

# ma plot
HEADING=paste(NAME, cnt_deg, "DGEs", cnt_up, "up", cnt_dn, "dn")
plot(log2(dge$baseMean),dge$log2FoldChange,main=HEADING,col="gray",pch=19,cex=0.5,xlab="log2(base mean)",ylab="log2(fold change)")
points(log2(sig$baseMean),sig$log2FoldChange,col="red",pch=19,cex=0.5)

# heatmap
if (cnt_deg>50) {cnt_deg=50}
if (cnt_deg>2) {
  mycols <- gsub("h20","yellow",ss1$trt)
  mycols <- gsub("mitoQ","orange",mycols)
  colfunc <- colorRampPalette(c("blue", "white", "red"))
  x1 <- x1/colSums(x1)*1e6
  mx <- as.matrix(x1[which(rownames(x1) %in% rownames(dge)[seq_len(cnt_deg)]),])
  heatmap.2(mx,trace="none",scale="row",margin=c(10,15),
    ColSideColors =mycols , col=colfunc(25),cexRow = 0.7 )
}

```

## contrast 6 test for EtOH vs MitoG in Het mice


```{r,de6,fig.height=7,fig.width=7}
NAME="DE6: EtOH vs mitoG in het mice"
ss1 <- subset(ss,geno==1)
ss1 <- subset(ss1,trt=="etoh" | trt=="mitoG")
ss1$trt <- factor(ss1$trt,levels=c("etoh","mitoG"))

x1 <- as.data.frame(xx[,ss1[,1]  ])
x1 <- x1[which(rowSums(x1)>10),]

dds <- DESeqDataSetFromMatrix(countData = x1 , colData = ss1, design = ~ trt )
res <- DESeq(dds)
z<- results(res)
vsd <- vst(dds, blind=FALSE)
zz<-cbind(as.data.frame(z),assay(vsd))
dge<-as.data.frame(zz[order(zz$pvalue),])
head(dge,10)
de6 <- dge

sig <- subset(dge,padj<0.05)
sig_up <- rownames(subset(sig,log2FoldChange>0))
sig_dn <- rownames(subset(sig,log2FoldChange<0))
cnt_deg <- nrow(sig)
cnt_up <-length(sig_up)
cnt_dn <- length(sig_dn)

#MDS PLOT
plot(cmdscale(dist(t(x1))), xlab="Coordinate 1", ylab="Coordinate 2", type = "n")
text(cmdscale(dist(t(x1))), labels=colnames(x1), )

# ma plot
HEADING=paste(NAME, cnt_deg, "DGEs", cnt_up, "up", cnt_dn, "dn")
plot(log2(dge$baseMean),dge$log2FoldChange,main=HEADING,col="gray",pch=19,cex=0.5,xlab="log2(base mean)",ylab="log2(fold change)")
points(log2(sig$baseMean),sig$log2FoldChange,col="red",pch=19,cex=0.5)

# heatmap
if (cnt_deg>50) {cnt_deg=50}
if (cnt_deg>2) {
  mycols <- gsub("etoh","yellow",ss1$trt)
  mycols <- gsub("mitoG","orange",mycols)
  colfunc <- colorRampPalette(c("blue", "white", "red"))
  x1 <- x1/colSums(x1)*1e6
  mx <- as.matrix(x1[which(rownames(x1) %in% rownames(dge)[seq_len(cnt_deg)]),])
  heatmap.2(mx,trace="none",scale="row",margin=c(10,15),
    ColSideColors =mycols , col=colfunc(25),cexRow = 0.7 )
}

```


## Contrast 7 test for WT v HET in H2O

```{r,de7,fig.height=7,fig.width=7}
NAME="DE7: wt vs het mice (H2O group)"
ss1 <- subset(ss,trt=="h2o" | trt=="h20")

x1 <- as.data.frame(xx[,ss1[,1]  ])
x1 <- x1[which(rowSums(x1)>10),]

dds <- DESeqDataSetFromMatrix(countData = x1 , colData = ss1, design = ~ geno )
res <- DESeq(dds)
z<- results(res)
vsd <- vst(dds, blind=FALSE)
zz<-cbind(as.data.frame(z),assay(vsd))
dge<-as.data.frame(zz[order(zz$pvalue),])
head(dge,10)
de7 <- dge

sig <- subset(dge,padj<0.05)
sig_up <- rownames(subset(sig,log2FoldChange>0))
sig_dn <- rownames(subset(sig,log2FoldChange<0))
cnt_deg <- nrow(sig)
cnt_up <-length(sig_up)
cnt_dn <- length(sig_dn)

#MDS PLOT
plot(cmdscale(dist(t(x1))), xlab="Coordinate 1", ylab="Coordinate 2", type = "n")
text(cmdscale(dist(t(x1))), labels=colnames(x1), )

# ma plot
HEADING=paste(NAME, cnt_deg, "DGEs", cnt_up, "up", cnt_dn, "dn")
plot(log2(dge$baseMean),dge$log2FoldChange,main=HEADING,col="gray",pch=19,cex=0.5,xlab="log2(base mean)",ylab="log2(fold change)")
points(log2(sig$baseMean),sig$log2FoldChange,col="red",pch=19,cex=0.5)

# heatmap
if (cnt_deg>50) {cnt_deg=50}
if (cnt_deg>2) {
  mycols <- gsub("0","yellow",ss1$geno)
  mycols <- gsub("1","orange",mycols)
  colfunc <- colorRampPalette(c("blue", "white", "red"))
  x1 <- x1/colSums(x1)*1e6
  mx <- as.matrix(x1[which(rownames(x1) %in% rownames(dge)[seq_len(cnt_deg)]),])
  heatmap.2(mx,trace="none",scale="row",margin=c(10,15),
    ColSideColors =mycols , col=colfunc(25),cexRow = 0.7 )
}

```

## Contrast 8 test for WT v HET in EtOH

```{r,de8,fig.height=7,fig.width=7}
NAME="DE8: wt vs het mice (EtOH group)"
ss1 <- subset(ss,trt=="etoh")

x1 <- as.data.frame(xx[,ss1[,1]  ])
x1 <- x1[which(rowSums(x1)>10),]

dds <- DESeqDataSetFromMatrix(countData = x1 , colData = ss1, design = ~ geno )
res <- DESeq(dds)
z<- results(res)
vsd <- vst(dds, blind=FALSE)
zz<-cbind(as.data.frame(z),assay(vsd))
dge<-as.data.frame(zz[order(zz$pvalue),])
head(dge,10)
de8 <- dge

sig <- subset(dge,padj<0.05)
sig_up <- rownames(subset(sig,log2FoldChange>0))
sig_dn <- rownames(subset(sig,log2FoldChange<0))
cnt_deg <- nrow(sig)
cnt_up <-length(sig_up)
cnt_dn <- length(sig_dn)

#MDS PLOT
plot(cmdscale(dist(t(x1))), xlab="Coordinate 1", ylab="Coordinate 2", type = "n")
text(cmdscale(dist(t(x1))), labels=colnames(x1), )

# ma plot
HEADING=paste(NAME, cnt_deg, "DGEs", cnt_up, "up", cnt_dn, "dn")
plot(log2(dge$baseMean),dge$log2FoldChange,main=HEADING,col="gray",pch=19,cex=0.5,xlab="log2(base mean)",ylab="log2(fold change)")
points(log2(sig$baseMean),sig$log2FoldChange,col="red",pch=19,cex=0.5)

# heatmap
if (cnt_deg>50) {cnt_deg=50}
if (cnt_deg>2) {
  mycols <- gsub("0","yellow",ss1$geno)
  mycols <- gsub("1","orange",mycols)
  colfunc <- colorRampPalette(c("blue", "white", "red"))
  x1 <- x1/colSums(x1)*1e6
  mx <- as.matrix(x1[which(rownames(x1) %in% rownames(dge)[seq_len(cnt_deg)]),])
  heatmap.2(mx,trace="none",scale="row",margin=c(10,15),
    ColSideColors =mycols , col=colfunc(25) ,cexRow = 0.7   )
}

```

## MitoG Venn diagrams

To determine whether mitoG attenuates diabetes we can overlap the DGEs found in these contrasts.
Diabetes is measured in contrast 8 and mitoG in contrast 6.

```{r,venn1}

de6_up <- rownames(subset(de6,padj<0.05 & log2FoldChange>0))
de6_dn <- rownames(subset(de6,padj<0.05 & log2FoldChange<0))
de8_dn <- rownames(subset(de8,padj<0.05 & log2FoldChange<0))
de8_up <- rownames(subset(de8,padj<0.05 & log2FoldChange>0))

v1 <- list("mitoG up"= de6_up, 
          "mitoG dn"= de6_dn,
	  "diabetes up"= de8_up,
          "dibetes dn"= de8_dn)

plot(euler(v1),quantities = TRUE)

```

## MitoG Mitch

Orthology file was downloaded from Ensembl Biomart Ensembl version 101.

Reactome sets obteined 8/Oct/2020.

```{r,mitch1}

mart <- read.table("mart_export.txt",fill=TRUE,sep="\t",header=TRUE)
gt <-  data.frame(mart$Mouse.gene.stable.ID,mart$Gene.name) 
gt <- gt[which(gt[,1]!=""),]

rownames(de6) <- sapply(strsplit(rownames(de6),"\\."),"[[",1) 
rownames(de8) <- sapply(strsplit(rownames(de8),"\\."),"[[",1) 

#download.file("https://reactome.org/download/current/ReactomePathways.gmt.zip", destfile="ReactomePathways.gmt.zip") 
#unzip("ReactomePathways.gmt.zip")
genesets <- gmt_import("ReactomePathways.gmt")

l <- list("diabetes"=de8,"mitoG"=de6)

m <- mitch_import(l, DEtype="deseq2",geneTable=gt)

# significance
res <- mitch_calc(m,genesets,priority="significance")
head(res$enrichment_result,20)
mitch_plots(res,"mitch_sig.pdf")
unlink("mitch_sig.html")
mitch_report(res,"mitch_sig.html")

# effect size
res <- mitch_calc(m,genesets,priority="effect")
head(res$enrichment_result,20)
mitch_plots(res,"mitch_eff.pdf")
unlink("mitch_eff.html")
mitch_report(res,"mitch_eff.html")

```

Gene sets were obtained from Msigdb version 7.2.
```{r,setsofinterest}
goi <- gmt_import("genesetsofinterest.gmt")

# significance
res <- mitch_calc(m,goi,priority="significance")
head(res$enrichment_result,20)
mitch_plots(res,"mitch_goi_sig.pdf")
unlink("mitch_goi_sig.html")
mitch_report(res,"mitch_goi_sig.html")

# effect size
res <- mitch_calc(m,goi,priority="effect")
head(res$enrichment_result,20)
mitch_plots(res,"mitch_goi_eff.pdf")
unlink("mitch_goi_eff.html")
mitch_report(res,"mitch_goi_eff.html")

```

## Session information

```{r,sessioninfo}

sessionInfo()

```
